packages/@atjson/renderer-hir/src/index.ts

```diff
@@ -153,16 +158,34 @@ export default class Renderer {
     );
   }
 
+  render(
+    document: Document | { text: string; marks: Mark[]; blocks: Block[] }
+  ) {
+    const [serializedDocument] = extractSlices(
+      document instanceof Document ? serialize(document) : document
+    );
+    return compile(
+      this,
+      null,
+      createTree(serializedDocument) as Map<
+        string,
+        Array<Block | Mark | string>
+      >,
+      ROOT,
+      {
+        document: serializedDocument as {
+          text: string;
+          marks: Mark[];
+          blocks: Block[];
+        },
+      }
+    );
+  }
+
   private slices: Map<string, { text: string; marks: Mark[]; blocks: Block[] }>;
 
   // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
-  constructor(
-    _document:
-      | Document
-      | { text: string; marks: Mark[]; blocks: Block[] }
-      | null,
-    ..._args: any[]
-  ) {
+  constructor(_document: RendererDocument, ..._args: any[]) {
```

Comment:
just a note here: prettier was clashing here with the fomratting which made it impossible to handle the eslint-disable-next-line